cli/src/host_perf_check.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017-2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::error::PerfCheckError;
+use log::info;
+use nix::unistd;
+use polkadot_node_core_pvf::sp_maybe_compressed_blob;
+use sc_service::BasePath;
+use service::kusama_runtime;
+use std::{
+	fs::{self, OpenOptions},
+	io::{self, Read, Write},
+	path::Path,
+	time::{Duration, Instant},
+};
+
+fn is_perf_check_done(path: &Path) -> io::Result<bool> {
+	let host_name_max_len = unistd::SysconfVar::HOST_NAME_MAX as usize;
+
+	let mut host_name = vec![0u8; host_name_max_len];
+	let mut buf = host_name.clone();
+	unistd::gethostname(&mut host_name).map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
+
+	let file = match fs::File::open(path) {
+		Ok(file) => file,
+		Err(err) if err.kind() == io::ErrorKind::NotFound => return Ok(false),
+		Err(err) => return Err(err),
+	};
+	let mut reader = io::BufReader::new(file);
+
+	reader.read_exact(&mut buf)?;
+
+	Ok(host_name == buf)
+}
+
+fn save_check_passed_file(path: &Path) -> io::Result<()> {
+	let host_name_max_len = unistd::SysconfVar::HOST_NAME_MAX as usize;
+	let mut host_name = vec![0u8; host_name_max_len];
+	unistd::gethostname(&mut host_name).map_err(|err| io::Error::new(io::ErrorKind::Other, err))?;
```

Comment:
This still needs verification that it works on Mac OS X, if unsure use the `hostname` crate.

Reply:
Both do `unsafe { libc::gethostname(ptr, len) };` under the hood, if it doesn't work on MacOS (which is unlikely), changing the crate won't help.

Reply:
:+1: that's all I wanted you to check.